src/agent/src/rpc.rs

```diff
@@ -1020,12 +1049,21 @@ impl protocols::agent_ttrpc::AgentService for AgentService {
         let mut sandbox = s.lock().await;
         // destroy all containers, clean up, notify agent to exit
         // etc.
-        sandbox.destroy().await.unwrap();
+        sandbox
+            .destroy()
+            .await
+            .map_err(|e| ttrpc_error(ttrpc::Code::INTERNAL, e.to_string()))?;
         // Close get_oom_event connection,
         // otherwise it will block the shutdown of ttrpc.
         sandbox.event_tx.take();
 
-        sandbox.sender.take().unwrap().send(1).unwrap();
+        sandbox
+            .sender
+            .take()
+            .ok_or("failed to get sandbox")
+            .map_err(|e| ttrpc_error(ttrpc::Code::INTERNAL, e.to_string()))?
```

Comment:
Also, you can collapse the `.ok_or(...).map_err(...)` chain into just an `.ok_or_else()`.